### PR TITLE
Hotfix/multiparted files

### PR DESCRIPTION
**Pull request by iBims1JFK**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -161,50 +157,59 @@ app.post('/api/saveUserImage', upload.single('data'), (req, res) => {
...
-    res.status(400)
-    res.send({ error: 'no file' })
-    return
-  }
+app.post('/api/writeData', upload.single('sensor'), async (req, res) => {
+  // if (!req.files) {
+  //   res.status(400)
+  //   res.send({ error: 'no file' })
+  //   return
+  // }
```
Comment (remunds) [inline app.js:187]: remove comment

File: app.js [hunk 2]
```diff
@@ -161,50 +157,59 @@ app.post('/api/saveUserImage', upload.single('data'), (req, res) => {
...
   await fs.write(options, readStream, async (error, file) => {
-    if (!error) {
-      res.status(200).send(req.body)
-      unlinkSync(req.files.sensor.tempFilePath, (err) => {
-        if (err) {
-          console.log('something went wrong')
-        }
-      })
+    if (error) {
+      res.status(500).send()
```
Comment (remunds) [inline app.js:200]: add error message

File: app.js [hunk 3]
```diff
@@ -161,50 +157,59 @@ app.post('/api/saveUserImage', upload.single('data'), (req, res) => {
...
     }
     console.log('wrote file with id: ' + file._id)
     // add the field downloads to file and chunks; add timestamp to chunk
     File.findByIdAndUpdate(file._id, { downloads: 0 }).exec()
     Chunk.updateMany({ files_id: file._id }, { downloads: 0, timestamp: Date.now() }).exec()
+    unlink(req.file.path, (err) => {
+      if (err) {
+        res.status(500).send()
+      } else {
+        res.status(200).send()
```
Comment (remunds) [inline app.js:210]: add error

**Review by remunds — CHANGES_REQUESTED** [review]
please rebase

**Review by remunds — DISMISSED** [review]
very nice!

**Review by laura-jehn — APPROVED** [review]
changes approved

**Review by laura-jehn — APPROVED** [review]
changes approved

**Pull request merged**